lib/hunter/commands/send.rb

```diff
@@ -66,25 +66,40 @@ def run
 
           request.body = data.to_json
 
-          begin
-            response = http.request(request)
-            response.value
-            puts "Successful transmission"
-          rescue Errno::ECONNREFUSED => e
-            puts "The server is unavailable"
-            puts e.message
-          rescue Net::HTTPServerException => e
-            if response.code == "401"
-              raise "Authentication key mismatch"
-            else
-              raise "Unknown HTTP error"
+          loop do
+            response = send_request(http, request)
+            if !@options.retry || response&.code == "200"
```

Comment:
What happens if we enable `--retry` and an unsuccessful status like (`401 Unauthorized`) is returned?